### PR TITLE
Fix regression with custom arguments being dropped in non-local executions

### PR DESCRIPTION
**Pull request by ionelmc**
This fixes regression from https://github.com/pytest-dev/pytest-xdist/pull/448 but lets see if it breaks other stuff.

### CODE REVIEW

**Comment by nicoddemus** [comment]
Thanks @ionelmc! It seems #448 unintentionlly also threw away pytest-xdist's *own* options, correct?

Any chance of writing an acceptance test that reproduces the problem to avoid future regressions (like this one in fact)?

We can ignore the py34 failures on Windows for now, it is unrelated and should be fixed/investigated seperately.

**Comment by ionelmc** [comment]
@nicoddemus I extended the chdir test to cover the hook problem.

**Comment by ionelmc** [comment]
Also, I didn't change the CHANGELOG.rst cause I don't really understand what that function does, or what the implications are.

**Comment by nicoddemus** [comment]
> Also, I didn't change the CHANGELOG.rst cause I don't really understand what that function does, or what the implications are.

No worries, thanks a lot for tackling this, we appreciate it. 👍 

I will take a look later.

File: tox.ini [hunk 1]
```diff
@@ -11,6 +11,7 @@ deps =
   pytestlatest: pytest
   pytestmaster: git+https://github.com/pytest-dev/pytest.git@master
   pytestfeatures: git+https://github.com/pytest-dev/pytest.git@features
+  -etesting/foobar
```
Comment (RonnyPfannschmidt) [inline tox.ini]: i  understand the need for this hack, please add a comment linking this fun hack with the test id

**Comment by RonnyPfannschmidt** [comment]
also extra note, i caught a glimpse of the real issue and i recoil thinking about how to make it possible to fix it

the config init keeps giving and giving

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt hmm even updating `virtualenv` didn't help. Any other ideas? I'm fine with removing py34 testing from Windows if we can't figure this out.

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus based on the errors, its using a different virtualenv than the one we have gotten installed, due to path issues, i believe its an indication the builds scripts are fundamentally wrong, if we take off py34, we should take it off everywhere

**Comment by nicoddemus** [comment]
>  i believe its an indication the builds scripts are fundamentally wrong, if we take off py34

Which "build scripts" you mean? Note that we test using plain tox, without intermediate scripts.

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus i looke in the wrong place, but as far as i can tell we dont update virtualenv

it might help to use tox-venv

**Pull request merged**